go/vt/sqlparser/sql.y

```diff
@@ -565,13 +575,13 @@ create_statement:
     // Change this to an alter statement
     $$ = &DDL{Action: AlterStr, Table: $7}
   }
-| CREATE VIEW table_name ddl_skip_to_end
+| CREATE VIEW table_name AS select_statement
   {
-    $$ = &DDL{Action: CreateStr, Table: $3.ToViewName()}
+    $$ = &DDL{Action: CreateStr, View: $3.ToViewName(), ViewExpr: $5}
   }
-| CREATE OR REPLACE VIEW table_name ddl_skip_to_end
+| CREATE OR REPLACE VIEW table_name AS select_statement
```

Comment:
Too much to ask you to add a or_replace_opt matcher?

Reply:
This ended up needing to touch the `constraint_opt` production, but I think the change there is a strict improvement for our use case. TAL.